core/src/main/java/org/apache/calcite/sql/validate/SqlConformanceEnum.java

```diff
@@ -314,4 +314,18 @@ public boolean allowExtendedTrim() {
     }
   }
 
+  @Override public boolean allowQualifyingCommonColumn() {
+    switch (this) {
+    case ORACLE_10:
+    case ORACLE_12:
+    case STRICT_92:
+    case STRICT_99:
+    case STRICT_2003:
+      return false;
```

Comment:
SQL standard also disallows qualifying the common columns[1].

[1] https://dev.mysql.com/doc/refman/8.0/en/join.html